libr/util/idpool.c

```diff
@@ -163,6 +162,54 @@ R_API void* r_id_storage_get(RIDStorage* storage, ut32 id) {
 	return storage->data[id];
 }
 
+R_API bool r_id_storage_get_lowest(RIDStorage *storage, ut32 *id) {
+	r_return_val_if_fail (storage, false);
+	size_t i;
+	for (i = 0; i < storage->size && !storage->data[i]; i++);
+	*id = i;
+	return i < storage->size;
+}
+
+R_API bool r_id_storage_get_highest(RIDStorage *storage, ut32 *id) {
+	r_return_val_if_fail (storage, false);
+	size_t i = 0;
+	if (storage->size > 0) {
+		for (i = storage->size - 1; !storage->data[i] && i > 0; i--);
+		*id = i;
+		return storage->data[i];  // != NULL;
+	}
+	*id = i;
+	return false;
+}
+
+R_API bool r_id_storage_get_next(RIDStorage *storage, ut32 *idref) {
+	r_return_val_if_fail (idref && storage && storage->data, false);
+	ut32 id = *idref;
+	if (storage->size < 1 || id >= storage->size || id == 0) {
```

Comment:
why return false if `id == 0`?

Reply:
because id=0 is treated as an invalid id in other places. there are no tests for this, but its the case. and i dont think this is the place to discuss/address this. so i just put it there for consistency with current behaviour

Reply:
0 is a valid id in fact

Reply:
@trufae Could you point to where id=0 is treated as invalid id?

Reply:
i removed that assumption. see r_oids_to_rear for example in uses id=0 to detect the loop has failed to find a match, cant find that ref anywhere else. but still i think i need to do further reviews and cleanups on all this API. but i dont think this PR is the place to do it